unlock-app/src/__tests__/services/web3Service.test.js

```diff
@@ -188,6 +189,81 @@ describe('Web3Service', () => {
     })
   })
 
+  describe('request enabling access to account', () => {
+    const netVersion = Math.floor(Math.random() * 100000)
+    let enable
+
+    beforeEach(done => {
+      const { providers } = configure(global)
+      nock.cleanAll()
+
+      enable = providers.HTTP.enable = jest.fn(() => Promise.resolve())
+      web3Service = new Web3Service(providers)
+      netVersionAndYield(netVersion)
+
+      UnlockContract.networks = {
+        [netVersion]: {
+          events: {},
+          links: {},
+          address: '0x3f313221a2af33fd8a430891291370632cb471bf',
+          transactionHash:
+            '0x8545541749873b42c96e1699c2e62f0f4062ca57f3398270423c1089232ef7dd',
+        },
+      }
+
+      web3Service.on('network.changed', () => {
```

Comment:
Could you use `once` here? This would ensure that if one reason or another this is called twice this would fail.